### ISSUE TITLE
[Feature Request] Support Field references? 

### ISSUE THREAD

**Issue by cdsimpkins** [keeweb/keeweb#296]
In the old Keepass Desktop client, it supported the idea of referencing another field for values.  This can be useful in environments where you want to logically separate systems that authenticate using the same credentials (think Active Directory authentication).  Are there any plans to support field references to other entries? 

<img width="255" alt="screen shot 2016-08-03 at 10 30 12 am" src="https://cloud.githubusercontent.com/assets/2373603/17369228/7c9ff2ae-5965-11e6-9395-ff7db2b96dc2.png">

**Comment by antelle** [comment]
No, there are no plans to support them at the moment. At least, we'll not display UUIDs in the user interface, this is insane. Maybe later we'll find another solution, sort of shortcuts or something like that.

**Comment by cdsimpkins** [comment]
It's not super important to me, I should really clean out old unused accounts, anyway.  But I know that the Desktop app had a convenient "picker" interface where you could search for an entry and choose what field to reference.  The user never had to actually deal with UUIDs unless they wanted to get into the guts of the database and explore.

That screenshot is what one of these entries in my old database looks like now in keeweb.

**Comment by antelle** [comment]
Hm, how did you get text in referenced fields?
When I insert a field reference in keepass, I get this:
![at](https://cloud.githubusercontent.com/assets/633557/17377310/d63fd962-59c1-11e6-923c-7e6e2d9b58a7.png)

But maybe I'm missing something.
Yes, when we support field references (I think eventually they will be supported), there will be picker interface as well, and text should also be displayed as a text field with some indication that it's a reference.

**Comment by cdsimpkins** [comment]
Sorry, I think I was unclear.  I think it inserted the UUID reference into the database in the username password field for the "new" entry.  I feel like the last version I used (maybe 2.27?) showed the referenced username in the entry list, but showed the UUID value in the entry inspection view.  There might also be another column you can turn on to that effect in the entry list, but I don't remember.

My database was created with KeePass 2.x years ago and I abandoned KeePass when I got a mac earlier this year.  These UUID references are just relics from that time, but I have a few of them.

Also, doesn't the preview window on the entry list say something like "username - {REF:....}" ?

Anyway, I'm just glad to know it's somewhere on radar.  It's not stopping me from doing anything, it's just a nice convenience feature.

**Comment by antelle** [comment]
Yes, I understand that it's a useful feature that worth implementing (that's why we'll keep the issue open) but it requires more work than just inserting `{ref:uuid}` to the field. So we'll support it once we are able to implement it completely, without adding ids. Now the feature needs prototyping...

**Comment by antelle** [comment]
**Status update**

Limited field references support was implemented:
- ⭐ databases containing field references will be opened with simple field references resolved to field values in the UI (i.e. references to id, which could be added automatically by KeePass)
- ⭐ the user will not see that it's a field reference
- ⭐ if you edit a field with reference, it's edited and saved as a simple value
- 💣 updating a referenced entry doesn't trigger updates of dependent entries while the db is open
- ⭐ editing another field in the entry containing field reference doesn't replace that reference with value, so we try to preserve references as much as we can
- 💣 there's no indication that it's a field reference

**Comment by cdsimpkins** [comment]
I like it.  

For the 3rd point, is this refreshed every time i unlock the db?  or every time i open the file?
For the 5th point, is there a way to add a text flag on the entry that says something like, "References fields from {list of entry names}"?  or maybe highlight the field name a different color if it's a reference?

Just spitballing.  Everything else sounds fair.  Good work!

**Comment by antelle** [comment]
They are refreshed every time you unlock the db, yes.
Exactly, we'll show an icon on a field that it's a field reference (like arrow on shortcut in windows), with a hint that it's a reference, by clicking it you will go to the entry which is referenced. It's just not matching current model, so a bit tricky to implement, that's why it will be implemented a bit later, not in v1.3 which should happen soon.

**Comment by toby3d** [comment]
Any progress at this moment? 😶

**Comment by antelle** [comment]
No. Field references are in rather good state for now. Adding new references and managing them is not a top priority.

**Comment by robswinkels** [comment]
Well, this description looked very hopeful but my password field uses references like {REF:P@I:xxx} but they are treated as text and most of the functionality described above does not work. I also cannot find it in the documentation, hence my remark here.

**Comment by Quentin-ctrL** [comment]
any updates ? 
It will be helpful to support field references. 
It's very helpful to keep a kdbx clean and open directly some URLs
https://keepass.info/help/base/placeholders.html#entryfields
https://keepass.info/help/base/fieldrefs.html
